### PR TITLE
Implement new `gateway feature: modify_recurring_payments`

### LINKED ISSUE #2173: Implement new gateway feature: `modify_recurring_payments`
See related: https://github.com/gocodebox/lifterlms-gateway-paypal/issues/52

## Goal

Allow payment gateways to determine if recurring payment details (such as trial end date, next payment date, etc...) can be modified via the admin panel order UI.

The new gateway feature can be added to the [supports](https://github.com/gocodebox/lifterlms/blob/81b1fff19d7b8a97e6a5d2780f800212050c6e1f/includes/abstracts/abstract.llms.payment.gateway.php#L101-L109) interface for a new gateway feature, probably called "modify_recurring_payments".

I think that this feature should be enabled by default *when* `recurring_payments` is `true`. We don't currently have a concept of a conditional feature so we'll probably need to reengineer the existing supports/feature functionality to accommodate a conditional feature default.

Ideally this should be implemented in a way that *does not* mess with existing gateways (which can edit by default)

### PR DESCRIPTION
**Pull request by eri-trabiccolo**
## Description
Fixes #2173

## How has this been tested?
manually (unit tests on the way)

## Types of changes
New feature (non-breaking change which adds functionality)

## Checklist:
- [x] My code has been tested.
- [x] My code passes all existing automated tests. 
- [x] My code follows the LifterLMS Coding & Documentation Standards.

### CODE REVIEW

**Comment by eri-trabiccolo** [comment]
@thomasplevy 
So I did as discussed:
https://github.com/gocodebox/lifterlms-gateway-paypal/issues/52#issuecomment-1148825190
but I kept https://github.com/gocodebox/lifterlms/blob/1579e1185bc406e6d0dab0464ec4a35e6e2016ed/includes/models/model.llms.order.php#L1884
I think it makes sense to keep it, as that it's both a gateway and order "support".

Fine for you?

**Comment by thomasplevy** [comment]
@eri-trabiccolo yea that makes sense

File: includes/admin/post-types/meta-boxes/class.llms.meta.box.order.submit.php [hunk 1]
```diff
@@ -136,6 +137,13 @@ public function save( $post_id ) {
 			'_llms_date_access_expires',
 		);
 
+		// Do not save recurring payments related dates if order's gateway do not support recurring payments modification.
+		if ( ! $order->supports_modify_recurring_payments() ) {
+			$editable_dates = array(
+				'_llms_date_access_expires',
+			);
+		}
+
```
Comment (thomasplevy) [inline includes/admin/post-types/meta-boxes/class.llms.meta.box.order.submit.php:142]: I'd rather see the array start as the dates that are always editable (access_expires) and then add the conditional ones later if the order supports recurring payments in favor of overwriting / redeclaring the array with only the expires date when the order cannot modify recurring payments.

**Review by thomasplevy — CHANGES_REQUESTED** [review]
This looks good to me, a minor piece of feedback. I haven't thoroughly reviewed all of the tests but the code seems solid.

File: includes/admin/post-types/meta-boxes/class.llms.meta.box.order.submit.php (hunk 1 above)
Comment (eri-trabiccolo) [reply includes/admin/post-types/meta-boxes/class.llms.meta.box.order.submit.php:142]: Ok, makes sense.

**Pull request merged**